### PR TITLE
Feature/find on shelf changes

### PR DESCRIPTION
**Pull request by Adamik10**
#### Link to issue
https://reload.atlassian.net/browse/DDFLSBP-712

#### Description
This PR adds the author name to the end of the Find on shelf modal from the nameSort property that exists on the Creators entity.
In order to get TS to be happy, I had to cast into one specific function. Let me know if you see a better way to deal with that, much appreciated 😁

#### Screenshot of the result
-

#### Additional comments or questions
ALSO UPDATE DPL DESIGN SYSTEM BEFORE MERGING THIS 
https://github.com/danskernesdigitalebibliotek/dpl-design-system/pull/668

### CODE REVIEW

File: src/apps/material-search/MaterialSearchList.tsx [hunk 1]
```diff
@@ -58,7 +59,7 @@ const MaterialSearchListResults: FC<MaterialSearchListResultsProps> = ({
       </div>
       <ol className="material-search-list__results">
         {data.map((work, index) => {
-          const authors = flattenCreators(work.creators);
+          const authors = flattenCreators(work.creators as Work["creators"]);
```
Comment (spaceo) [inline src/apps/material-search/MaterialSearchList.tsx:62]: I think we should try to infer the `Work` type as early as possible so we do not need to use type assertions later on.
Please look at this [suggestion](https://stackblitz.com/edit/vitejs-vite-hzqbpm?file=src%2FMaterialSearchList.tsx%3AL54&terminal=dev)

File: src/components/find-on-shelf/FindOnShelfModalBody.tsx [hunk 2]
```diff
@@ -55,6 +57,10 @@ const FindOnShelfModalBody: FC<FindOnShelfModalBodyProps> = ({
     config
   });
   const author = creatorsToString(flattenCreators(authors), t);
+  // Inverted author - last name, then first name is used in find on shelf modal.
+  const invertedAuthor = capitalizeFirstLetters(
+    creatorsToString(flattenCreatorsLastNameFirst(authors), t)
```
Comment (spaceo) [inline src/components/find-on-shelf/FindOnShelfModalBody.tsx:62]: Nice way of composing existing functions 👍

File: src/core/utils/helpers/general.ts [hunk 3]
```diff
@@ -47,6 +54,11 @@ export const flattenCreators = (creators: Work["creators"]) =>
     return creator.display;
   });
 
+export const flattenCreatorsLastNameFirst = (creators: Work["creators"]) =>
+  creators.map((creator: Work["creators"][0]) => {
```
Comment (spaceo) [inline src/core/utils/helpers/general.ts:58]: I dont think it is necessary to define the creators type again. TS is capable of inferring that by itself:
```suggestion
  creators.map((creator) => {
```

**Review by spaceo — CHANGES_REQUESTED** [review]
I have some commentz...

File: src/apps/material-search/MaterialSearchList.tsx (hunk 1 above)
Comment (Adamik10) [reply src/apps/material-search/MaterialSearchList.tsx:62]: That's a great way to do it! thx for the suggestion ✨

**Review by spaceo — APPROVED** [review]
LGTM 👍

**Pull request merged**